src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-devices-selection-groups/osd-devices-selection-groups.component.ts

```diff
@@ -28,6 +29,8 @@ export class OsdDevicesSelectionGroupsComponent implements OnInit, OnChanges {
 
   @Input() canSelect: boolean;
 
+  @Input() clusterCreation = false;
+
```

Comment:
I was telling @aaSharma14 in the service creation PR #43072 that the inner/lower level components shouldn't contain (self-modify themselves based on) references of concerns of upper level instances (ClusterCreation workflow is a higher level/multi-component concern). The example I gave to him is [this one](https://github.com/ceph/ceph/pull/43072#discussion_r705569360).

Simply put, that means that the reusable/reused components (hosts, services, OSDs) should provide an interface flexible enough (yet generic) to do these customizations. If columns need to be hidden, instead of hiding them based on this ad-hoc clusterCreation param, we should expose a `columnsToHide` option.

If we don't follow this approach (called encapsulation), the next time we want to add a... let's call it "OSD-Replacement-Wizard", we'll to add another input argument called `osdCReplaceWizard` and implement lots of complex conditional logic with it (taking care of not breaking/interfering with the clusterCreation appearance).

On the other, if we make the inner component interface expressive enough to do what the callers need from that component, we can keep a simple/generic internal logic, fully decoupled from what's happening outside.

Reply:
Makes sense to me. Currently working on it. Made a small change in this commit https://github.com/ceph/ceph/pull/43041/commits/278d79ef58b1cf3bcac23bc5f5305cae7c50f4d2. Can you see if its in the right direction? @epuertat